code/modules/vore/eating/bellymodes_vr.dm

```diff
@@ -744,6 +769,214 @@
 						internal_contents -= P
 		return
 
+///////////////////////////// DM_TRANSFORM_REPLICA_EGG /////////////////////////////
+	if(digest_mode == DM_TRANSFORM_REPLICA_EGG && ishuman(owner))
+		for (var/mob/living/carbon/human/P in internal_contents)
+			if(P.stat)
+				continue
+			if(P.absorbed) //If they're absorbed, don't egg them
+				return
+			var/mob/living/carbon/human/O = owner
+
+			if (O.custom_species)
+				var/defined_species = O.custom_species
+				P.r_hair 			= O.r_hair
+				P.r_facial 			= O.r_facial
+				P.g_hair 			= O.g_hair
+				P.g_facial 			= O.g_facial
+				P.b_hair 			= O.b_hair
+				P.b_facial 			= O.b_facial
+				P.r_skin 			= O.r_skin
+				P.g_skin 			= O.g_skin
+				P.b_skin 			= O.b_skin
+				P.tail_style 		= O.tail_style
+				P.r_tail 			= O.r_tail
+				P.g_tail 			= O.g_tail
+				P.b_tail 			= O.b_tail
+				P.ear_style 		= O.ear_style
+				P.h_style 			= O.h_style //Since some things are required, like the cobra hood.
+				P.species 			= O.species
+				P.custom_species 	= O.custom_species
+				P.r_eyes 			= O.r_eyes
+				P.g_eyes 			= O.g_eyes
+				P.b_eyes 			= O.b_eyes
+
+				for(var/obj/item/organ/I in P.internal_organs) //This prevents organ rejection
+					I.species = O.species
+
+				for(var/obj/item/organ/external/chest/A in O.organs)
+					for(var/obj/item/organ/external/Z in P.organs) //This makes their limb sprites look correct.
+						Z.species = O.species
+						Z.h_col = A.h_col
+						Z.s_col = A.s_col
+				P << "<span class='notice'>You lose sensation of your body, feeling only the warmth around you as you're encased in an egg. </span>"
+				owner << "<span class='notice'>You shift as you make dramatic changes to your captive's body as you encase them in an egg.</span>"
+				P.fixblood()
+				P.update_hair()
+				P.update_body()
+				P.update_tail_showing()
+				P.update_eyes()
+				P.updateicon()
+				switch(O.egg_type)
+					if("Unathi")
+						var/obj/structure/closet/secure_closet/egg/unathi/J = new /obj/structure/closet/secure_closet/egg/unathi(O.loc)
+						P.forceMove(J)
```

Comment:
I'm usually opposed to telling people to fix things that were existing before they messed with the files in the PR, however you're making it worse with these, so please change these to not use the same duplicated code 8+ times in the same file. Either make egg types a list of "Species Name" = /path or something else.